From 936708b2d947917be5aed3de0b0de85ded2d3283 Mon Sep 17 00:00:00 2001
From: Ross Lagerwall <rosslagerwall@gmail.com>
Date: Thu, 7 Nov 2013 12:11:40 +0200
Subject: [PATCH 06/11] sftp: Fix handling of multiple reads of the packet
 length

In certain cases, reading the packet length may take more than one call.
Make this work by calculating the offset into the reply_size correctly.

https://bugzilla.gnome.org/show_bug.cgi?id=532951
---
 daemon/gvfsbackendsftp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/daemon/gvfsbackendsftp.c b/daemon/gvfsbackendsftp.c
index 23d5b00..2098fd5 100644
--- a/daemon/gvfsbackendsftp.c
+++ b/daemon/gvfsbackendsftp.c
@@ -1245,7 +1245,7 @@ read_reply_async_got_len  (GObject *source_object,
   if (backend->reply_size_read < 4)
     {
       g_input_stream_read_async (backend->reply_stream,
-				 &backend->reply_size + backend->reply_size_read, 4 - backend->reply_size_read,
+				 (char *)&backend->reply_size + backend->reply_size_read, 4 - backend->reply_size_read,
 				 0, backend->reply_stream_cancellable, read_reply_async_got_len,
 				 backend);
       return;
-- 
1.9.0

